lib/DataStore.js

```diff
@@ -120,6 +120,37 @@ DataStore.prototype.getRoom = function(roomId, ircDomain, ircChannel, origin) {
 };
 
 /**
+ * Get all Matrix <--> IRC room mappings from the database.
+ * @return {Promise} A promise which resolves to a map:
+ *      $roomId => ['server #channel1', 'server #channel2',...]
+ */
+DataStore.prototype.getAllMappings = Promise.coroutine(function*() {
+    let entries = yield this._roomStore.select(
+        {
+            matrix_id : {$exists: true},
+            remote_id: {$exists: true},
+        }
```

Comment:
Whilst this does what it says on the tin, this is almost certainly not what you want. This will pick up PM rooms which look like:
```json
{
"id":"PM_@kegan:matrix.org @mozilla_ffwywvmervtdliqphiabshsmibvfns:matrix.org",
"remote_id":"irc.mozilla.org ffwywvmervtdliqphiabshsmibvfns",
"matrix_id":"!gDHoKrNsohmsFydWRb:matrix.org",
"remote":{"domain":"irc.mozilla.org","type":"pm","channel":"ffwywvmervtdliqphiabshsmibvfns"},
"matrix":{"extras":{}},
"data":{"virtual_user_id":"@mozilla_ffwywvmervtdliqphiabshsmibvfns:matrix.org","real_user_id":"@kegan:matrix.org"},
"_id":"3z5eJBIRz6f8d8FK"
}
```

What you really want is to pick out all *channels*:
```json
{
"id":"!PcCidzjUKXHKImhTrB:matrix.org irc.mozilla.org #rust",
"remote_id":"irc.mozilla.org #rust",
"matrix_id":"!PcCidzjUKXHKImhTrB:matrix.org",
"remote":{"domain":"irc.mozilla.org","channel":"#rust","type":"channel"},
"matrix":{"extras":{}},
"data":{"origin":"config"},
"_id":"cmQ0M7lHHAZg76nQ"
}
```

AKA `remote.type === "channel"`